backend/src/main/java/atlantbh/restaurants/controllers/UserController.java

```diff
@@ -0,0 +1,37 @@
+package atlantbh.restaurants.controllers;
+
+import atlantbh.restaurants.controllers.dto.UserRegistrationDTO;
+import atlantbh.restaurants.models.DefaultError;
+import atlantbh.restaurants.models.ErrorResponseWrapper;
+import atlantbh.restaurants.models.User;
+import atlantbh.restaurants.services.UserService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+
+@RestController
+public class UserController extends BaseController<User, UserService> {
+
+    public ResponseEntity registerUser(@Valid @RequestBody UserRegistrationDTO user, BindingResult bindingResult) {
+        if(service.isEmailTaken(user.getEmail()) == true) {
+            DefaultError error = new DefaultError("email", "Email is already in use");
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(new ErrorResponseWrapper(error));
+        }
+        if(!user.getPassword().equals(user.getConfirmedPassword())) {
+            DefaultError error = new DefaultError("password", "Passwords don't match");
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(new ErrorResponseWrapper(error));
+        }
+        if (bindingResult.hasErrors()) {
+            DefaultError error = new DefaultError();
+            error.setField(bindingResult.getFieldError().getField());
+            error.setMessage(bindingResult.getFieldError().getDefaultMessage());
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(new ErrorResponseWrapper(error));
+        }
+        User registeredUser = service.register(user);
```

Comment:
```
return ResponseEntity.ok(service.register(user));
```

All of these checks can be on service layers.. e.g. email check in `UserService` can throw exception with custom message ("Email already in use"). Then you can catch exception in this controller and create badRequest response with custom messages for every of these cases.

Reply:
Moved business logic to service 92c4f7264b274d7da369717e0ed1478a93d30ee4